packages/ActionBar/test/spec/ActionBar.spec.js

```diff
@@ -0,0 +1,94 @@
+import testers from "../../src/hooks/useFilter/testers";
+import Filter from "../../src/components/Filter/Filter";
+
+import sort from "../../src/hooks/useSort/sort";
+import sampleData from "../data";
+
+describe("ActionBar Filter", () => {
+  it("should run filter rules IS NOT", () => {
+    expect(testers[Filter.rules.IS_NOT]("value", "")).toBe(true);
+    expect(testers[Filter.rules.IS_NOT]("value", "testValue")).toBe(true);
+    expect(testers[Filter.rules.IS_NOT]("value", "value")).toBe(false);
+  });
+
+  it("should run filter rules DOES NOT CONTAIN", () => {
+    expect(testers[Filter.rules.DOES_NOT_CONTAIN]("value", "test value")).toBe(true);
+    expect(testers[Filter.rules.DOES_NOT_CONTAIN]("test VaLue", "value")).toBe(false);
+  });
+
+  it("should run filter rules IS NOT BLANK", () => {
+    expect(testers[Filter.rules.IS_NOT_BLANK]("test value")).toBe(true);
+    expect(testers[Filter.rules.IS_NOT_BLANK]("")).toBe(false);
+  });
+  it("should run filter rules IS NOT EQUAL TO", () => {
+    expect(testers[Filter.rules.NOT_EQUAL_TO](100, 200)).toBe(true);
+    expect(testers[Filter.rules.NOT_EQUAL_TO](100, 100)).toBe(false);
+  });
+
+  it("should run filter rules IS GREATER THAN OR EQUAL TO ", () => {
+    expect(testers[Filter.rules.GREATER_THAN_OR_EQUAL_TO](200, 200)).toBe(true);
+    expect(testers[Filter.rules.GREATER_THAN_OR_EQUAL_TO](250, 200)).toBe(true);
+    expect(testers[Filter.rules.GREATER_THAN_OR_EQUAL_TO](150, 200)).toBe(false);
+  });
+  it("should run filter rules IS LESS THAN OR EQUAL TO ", () => {
+    expect(testers[Filter.rules.LESS_THAN_OR_EQUAL_TO](200, 200)).toBe(true);
+    expect(testers[Filter.rules.LESS_THAN_OR_EQUAL_TO](150, 200)).toBe(true);
+    expect(testers[Filter.rules.LESS_THAN_OR_EQUAL_TO](250, 200)).toBe(false);
+  });
+  it("should run filter rules IS NOT EMPTY", () => {
+    expect(testers[Filter.rules.IS_NOT_BLANK]("test value")).toBe(true);
+    expect(testers[Filter.rules.IS_NOT_BLANK]("")).toBe(false);
+  });
+  it("should run filter rules IS AFTER", () => {
+    expect(testers[Filter.rules.IS_AFTER]("10/08/2020", "10/02/2020", { momentParsingFormat: "MM/DD/YYYY" })).toBe(
+      true
+    );
+    expect(testers[Filter.rules.IS_AFTER]("10/02/2020", "10/08/2020", { momentParsingFormat: "MM/DD/YYYY" })).toBe(
+      false
+    );
+  });
+});

```

Comment:
Sorry I didn't post this yesterday cause I'm not sure which PR should address which part of test. I found these two chunks have some duplication
https://github.com/acl-services/paprika/pull/516/files#diff-d92522ffaac22ef6b206928dfe05277dR13-R58
https://github.com/acl-services/paprika/pull/517/files#diff-6df3961b0048fddc67514b35439bf671R7-R50
Maybe in your branch you should test`filterData` function in reducer instead of the helper functions? Cause it looks like you're writing most of the integration tests and Kaan is writing unit tests?
I'm not 100% sure though, please correct me
cc @KaanDarcey @nahumzs

Reply:
For the helpers, Kaan and I split up. I tested half of it, then Kaan did the other half. In terms of the filterData, what do you mean about _testfilterData function in reducer_? Can you please explain so we can add it.